uxarray/grid.py

```diff
@@ -225,3 +227,34 @@ def __init_ds_var_names__(self):
             "nMesh2_face": "nMesh2_face",
             "nMaxMesh2_face_nodes": "nMaxMesh2_face_nodes"
         }
+
+    def __init_ds_var_methods__(self):
+        """Initialize variables for directly accessing Coordinate and
+        Data variables through ugrid conventions
+
+        Examples
+        ----------
+        Initialize some Grid Object
+        >>> grid = ux.open_dataset(file_path)
+
+        Return Xarray Data Array
+        >>> x = grid.Mesh2_node_x      
+
+        Return Numpy Array
+        >>> x = grid.Mesh2_node_x.values
+        """
+    
+        # For Testing (caused other tests to fail)
+        if self.ds is None:
```

Comment:
I'd expect `ds` not being `None` to be guaranteed until this point, isn't it?

Reply:
How about replacing the warning in _shapefile.py with an exception like this:

`raise RuntimeError("Function not implemented yet. FYI, attempted to read SHAPE file: " + str(filepath))`

Then, calling the assertion lines in `test_read_shpfile` like this:

```
with self.assertRaises(RuntimeError):
            shp_filename = current_path / "meshfiles" / "grid_fire.shp"
            tgrid = ux.Grid(str(shp_filename))
```

I believe with thee, you could remove your `self.ds is None` check.